packages/config/src/options.ts

```diff
@@ -27,31 +29,35 @@ export type BreakingOptionErrorKey =
   | 'RENAMED_CONFIG_OPTION'
   | 'TEST_FILES_RENAMED'
 
-type TestingType = 'e2e' | 'component'
+// The test-time override levels
+export const ALL_OVERRIDE_LEVELS = ['code', 'event', 'suite', 'test', 'runtime'] as const
```

Comment:
I think it would be useful to have the comment above this explain what each level means

Reply:
I'm having trouble discerning the practical difference between 'code' and 'runtime'. It seems that 'code' is when setting config at the top/global level and 'runtime' is when it's inside a test, but it seems like the result is the same either way, since they're [handled the same way during validation](https://github.com/cypress-io/cypress/pull/23040/files#diff-f635b34a5d81758a8ce5c487afae2d4fb0668271687a8dd7775898afac2fd328R102). The only difference seems to be [when crafting the error message](https://github.com/cypress-io/cypress/pull/23040/files#diff-6c0564db103220b716eb75e4a31c25a8192ac068e816710298d2e14d80cf7499R274-R276), but it seems like that's just a nicety that could apply to the 'code' error as well. Is there a case where a config value would only be overrideable at 'code' level and not 'runtime' or vice versa? If not, it seems like they could be merged into the same value.

Reply:
So the code level is either when the support file or spec file is loaded before mocha has processed the spec to determine the suites/tests/runnable in the code. Originally I had named it "supportFile" but found it could be when either support file or spec file was loaded into the AUT.

I do think from a messaging standpoint the delineation is helpful since runTime is when a mocha runnable itself is executing, but maybe it's not? 🤷🏻‍♀️ are you confused on the difference between the two or are you wanting "slimmer" override levels and messaging?

Reply:
I understand the difference now - thanks for the clarification. If there's a use-case for having them separate, that's fine, but I would suggest renaming them to something like `runtime-global` and `runtime-test`. However, if there's not a use-case, I'd say YAGNI 😅 It's an internal API, so we can always change it later if a use-case emerges.